# Fake News Detection

In [2]:
# Importing the nevessary libraries
import numpy as np
import pandas as pd
import itertools
from sklearn.cross_validation import train_test_split
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.linear_model import PassiveAggressiveClassifier
from sklearn.metrics import accuracy_score, confusion_matrix

In [3]:
# Importing the dataset
dataset = pd.read_csv('news.csv')
X = dataset.text
y = dataset.label

In [4]:
# Splitting the dataset into the Training set and Test set
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.20, random_state = 0)

In [6]:
# Initialise and fit the TfidVectorizer
vectorizer = TfidfVectorizer(stop_words = 'english', max_df = 0.7)
X_train = vectorizer.fit_transform(X_train)
X_test = vectorizer.transform(X_test)

In [12]:
# Fitting PassiveAggressive Classifer to the Training set
pac = PassiveAggressiveClassifier()
pac.fit(X_train, y_train)

PassiveAggressiveClassifier(C=1.0, class_weight=None, fit_intercept=True,
              loss='hinge', n_iter=5, n_jobs=1, random_state=None,
              shuffle=True, verbose=0, warm_start=False)

In [16]:
# Predicting the Test set results
y_pred = pac.predict(X_test)
score=accuracy_score(y_test,y_pred)
print('Accuracy:', {round(score*100,2)})

Accuracy: {93.370000000000005}


In [18]:
# Making the Confusion Matrix
cm = confusion_matrix(y_test, y_pred)
cm

array([[569,  46],
       [ 38, 614]])

This is having 569 true positives and 614 true negatives.
And also 46 false positives and 38 false negatives.